lib/OpenQA/Utils.pm

```diff
@@ -349,20 +353,30 @@ sub run_cmd_with_log_return_error ($cmd, %args) {
             log_error("cmd returned $return_code");
         }
         return {
-            status => $ipc_run_succeeded,
+            status => $ipc_run,
             return_code => $return_code,
             stdout => $stdout,
             stderr => $stderr,
         };
     }
     catch {
-        return {
-            status => 0,
-            return_code => undef,
-            stderr => 'an internal error occurred',
-            stdout => '',
+        my $additional_info = "If you use http for CASEDIR or NEEDLES_DIR is adviced to
```

Comment:
`run_cmd_with_log_return_error` is a general purpose function not only used for git commands.
It doesn't make sense to add a message about git here.

Also, we shouldn't add lengthy directions into the error message, but link to the documentation, as it was suggested in the ticket